src/imitation/data/rollout.py

```diff
@@ -142,20 +148,26 @@ def add_steps_and_auto_finish(
             each `True` in the `dones` argument.
         """
         trajs: List[types.TrajectoryWithRew] = []
-        for env_idx in range(len(obs)):
+        wrapped_obs = types.DictObs.maybe_wrap(obs)
```

Comment:
Nit: does this belong in `DictObs`? A factory method that always returns `DictObs` would. Something that *sometimes* returns `DictObs` feels borderline.

Reply:
moved outside into a cluster of utility methods in types.py